tests/testsuite/git_auth.rs

```diff
@@ -209,25 +215,42 @@ fn https_something_happens() {
 
     p.cargo("check -v")
         .with_status(101)
-        .with_stderr_contains(&format!(
-            "[UPDATING] git repository `https://{addr}/foo/bar`"
-        ))
-        .with_stderr_contains(&format!(
+        .with_stderr_data(&format!(
             "\
+[UPDATING] git repository `https://{addr}/foo/bar`
+[ERROR] failed to get `bar` as a dependency of package `foo v0.0.1 ([ROOT]/foo)`
+
+Caused by:
+  failed to load source for dependency `bar`
+
 Caused by:
-  {errmsg}
+  Unable to update https://{addr}/foo/bar
+
+Caused by:
+  failed to clone into: [ROOT]/home/.cargo/git/db/bar-[HASH]
+
+Caused by:
+{errmsg}
 ",
             errmsg = if cargo_uses_gitoxide() {
-                "[..]SSL connect error [..]"
+                r"  network failure seems to have happened
+  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
+  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
+
+Caused by:
+  An IO error occurred when talking to the server
+
+Caused by:
+  [35] SSL connect error ([..])"
             } else if cfg!(windows) {
-                "[..]failed to send request: [..]"
+                "  failed to send request: A connection with the server could not be established\n  ; class=Os (2)"
             } else if cfg!(target_os = "macos") {
                 // macOS is difficult to tests as some builds may use Security.framework,
                 // while others may use OpenSSL. In that case, let's just not verify the error
                 // message here.
-                "[..]"
+                "..."
             } else {
-                "[..]SSL error: [..]"
+                r"  SSL [ERROR] syscall failure: ; class=Os (2)"
```

Comment:
ditto for platform-dependent error message 
```suggestion
                r"[..]SSL [ERROR][..]"
```